xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -2214,11 +2214,13 @@ private void handleResourceUpdate(
             }
             retainedResources.add(edsName);
           }
-          continue;
+        } else if (invalidResources.contains(resourceName)) {
```

Comment:
This looks incomplete, as the case when `type != ResourceType.LDS && type != ResourceType.CDS` will no longer trigger `onAbsent()`, ever.

Reply:
Yea I was confused as well it was existing behavior for other resource types. Looks a bug?
cc. @dapengzhang0

Reply:
For RDS,EDS `onAbsent()` is handled below  (`dependentSubscribers`)

https://github.com/grpc/grpc-java/pull/8660/files#diff-6b019eb1d3518381d336168847890513a235ea4974fcb3c76a2e9ace43f1d460R2233

@YifeiZhuang your change is only for LDS and CDS type.

Reply:
> This looks incomplete, as the case when type != ResourceType.LDS && type != ResourceType.CDS will no longer trigger onAbsent(), ever.

@ejona86  was your original question about [RDS and EDS](https://github.com/grpc/grpc-java/blob/a5c526c12f699af9a846d9faabc18038ef431675/xds/src/main/java/io/grpc/xds/ClientXdsClient.java#L1413) never gets `onAbsent()`? if so, then it looks by design and makes sense as Penn explained to me offline. They are SoW not incremental. 
Then I think the change is what we want.

Reply:
Oh, I see it. I mixed up the indentation.